Lecture6/src/test/java/TestOzBy.java

```diff
@@ -0,0 +1,83 @@
+import org.openqa.selenium.*;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+public class TestOzBy {
+    private WebDriver driver;
+    private final String URL = "https://www.oz.by";
+    private final String MY_EMAIL = "lebedevaolga021@gmail.com";
+    private final String MY_PASSWORD = "wPdAp9";
+
+    @BeforeClass
+    public void beforeClass() throws InterruptedException {
+        System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
+        driver = new ChromeDriver();
+    }
+
+    @AfterClass
+    public void afterClass() {
+        driver.quit();
+    }
+
+    @Test
+    private void loginSelectAndAddBookToBasket() throws InterruptedException {
+        driver.get(URL);
+        implicitWait();
+        loginToSite();
+        implicitWait();
+        searchBook();
+        implicitWait();
+        putInBasket();
+        checkBasket();
+        implicitWait();
+    }
+
+    public void loginToSite() throws InterruptedException, NoSuchElementException {
+        WebElement loginBar = driver.findElement(By.className("top-panel__userbar__auth"));
+        loginBar.click();
+        WebElement enterByEmail = driver.findElement(By.id("loginFormLoginEmailLink"));
+        enterByEmail.click();
+        WebElement emailField = driver.findElement(By.name("cl_email"));
+        emailField.sendKeys(MY_EMAIL);
+        WebElement passwordField = driver.findElement(By.xpath("//*[@id=\"loginForm\"]/div[2]/div[1]/div[2]/input"));
+        passwordField.sendKeys(MY_PASSWORD);
+        WebElement clickLogin = driver.findElement(By.cssSelector("#loginForm > button"));
+        clickLogin.click();
+    }
+
+    public void searchBook() throws StaleElementReferenceException {
```

Comment:
Could you please clarify why did you add `StaleElementReferenceException ` in method signature?